src/CPUActivityAgent.cpp

```diff
@@ -136,46 +138,61 @@ namespace geopm
-    void CPUActivityAgent::validate_policy(std::vector<double> &in_policy) const
+    void CPUActivityAgent::init_constconfig_io()
     {
-        GEOPM_DEBUG_ASSERT(in_policy.size() == M_NUM_POLICY,
-                           "CPUActivityAgent::" + std::string(__func__) +
-                           "(): policy vector not correctly sized.  Expected  " +
-                           std::to_string(M_NUM_POLICY) + ", actual: " +
-                           std::to_string(in_policy.size()));
+        m_qm_max_rate = {};
+        const auto ALL_NAMES = m_platform_io.signal_names();
 
-        // Check for NAN to set default values for policy
-        if (std::isnan(in_policy[M_POLICY_CPU_FREQ_MAX])) {
-            in_policy[M_POLICY_CPU_FREQ_MAX] = m_freq_core_max;
+        // F efficient values
+        std::string fe_constconfig = "CONST_CONFIG::CPU_FREQUENCY_EFFICIENT_HIGH_INTENSITY";
+        if (ALL_NAMES.count(fe_constconfig) != 0) {
+            m_freq_core_efficient = m_platform_io.read_signal(fe_constconfig, GEOPM_DOMAIN_BOARD, 0);
         }
-
-        // Check for NAN to set default values for policy
-        if (std::isnan(in_policy[M_POLICY_CPU_FREQ_EFFICIENT])) {
-            in_policy[M_POLICY_CPU_FREQ_EFFICIENT] = m_freq_core_min;
+        else {
+            m_freq_core_efficient = m_freq_core_min;
         }
 
-        //////////////////////////
-        //UNCORE POLICY CHECKING//
-        //////////////////////////
-        if (std::isnan(in_policy[M_POLICY_UNCORE_FREQ_MAX])) {
-            in_policy[M_POLICY_UNCORE_FREQ_MAX] = m_freq_uncore_max;
+        fe_constconfig = "CONST_CONFIG::CPU_UNCORE_FREQUENCY_EFFICIENT_HIGH_INTENSITY";
+        if (ALL_NAMES.count(fe_constconfig) != 0) {
+            m_freq_uncore_efficient = m_platform_io.read_signal(fe_constconfig, GEOPM_DOMAIN_BOARD, 0);
         }
-        if (std::isnan(in_policy[M_POLICY_UNCORE_FREQ_EFFICIENT])) {
-            in_policy[M_POLICY_UNCORE_FREQ_EFFICIENT] = m_freq_uncore_min;
+        else {
+            m_freq_uncore_efficient = m_freq_uncore_min;
         }
 
-        // When the policy is all NaNs, this check also verifies that
-        // the system was not left in a bad state with regard to the
-        // UNCORE_FREQUENCY_<MAX/MIN>_CONTROLs.
-        if (in_policy[M_POLICY_UNCORE_FREQ_EFFICIENT] > in_policy[M_POLICY_UNCORE_FREQ_MAX]) {
+        // Grab all (uncore frequency, max memory bandwidth) pairs
+        for (int entry_idx = 0; entry_idx < (int)ALL_NAMES.size(); ++entry_idx) {
+            const std::string KEY_NAME = "CONST_CONFIG::CPU_UNCORE_FREQUENCY_" +
+                                          std::to_string(entry_idx);
+            const std::string VAL_NAME = "CONST_CONFIG::CPU_UNCORE_MAX_MEMORY_BANDWIDTH_" +
+                                          std::to_string(entry_idx);
+            if (ALL_NAMES.find(KEY_NAME) != ALL_NAMES.end() &&
+                ALL_NAMES.find(VAL_NAME) != ALL_NAMES.end()) {
```

Comment:
Previously, you were using the ``count()`` method to search in the ``ALL_NAMES`` set, whereas here you're using the ``find()`` method. I recommend you remain consistent. Using ``find()`` seems more readable, but I like that using the ``count()`` method avoids creating a couple of iterators (and with set the count is going to be either 0 or 1). I have no strong preference, just remain consistent.

Reply:
My natural preference is find, which is why inconsistency cropped up, but we use .count() elsewhere in GEOPM so I'll convert to count()